internal/ascode/cache_util.go

```diff
@@ -0,0 +1,49 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package common
+
+import (
+	"errors"
+	"github.com/patrickmn/go-cache"
+	log "github.com/sirupsen/logrus"
+	"reflect"
+)
+
+type CacheGetter func(id string) (interface{}, error)
+
+func GetItemFromCache(c *cache.Cache, key string, fn CacheGetter) (interface{}, error) {
```

Comment:
why not just use a map and remove this and the dependencies? This "doing" something?